scripts/branching-urls/a1-root-allowed/pyproject.toml

```diff
@@ -0,0 +1,14 @@
+[project]
+name = "a"
+version = "0.1.0"
+description = "Test package for direct URLs in branches"
+dependencies = [
+  # Valid, disjoint split
+  "iniconfig @ https://files.pythonhosted.org/packages/9b/dd/b3c12c6d707058fa947864b67f0c4e0c39ef8610988d7baea9578f3c48f3/iniconfig-1.1.1-py2.py3-none-any.whl ; python_version < '3.12'",
+  "iniconfig @ https://files.pythonhosted.org/packages/ef/a6/62565a6e1cf69e10f5727360368e451d4b7f58beeac6173dc9db836a5b46/iniconfig-2.0.0-py3-none-any.whl ; python_version >= '3.12'",
+]
+requires-python = ">=3.11,<3.13"
+
+[build-system]
+requires = ["flit_core>=3.8,<4"]
+build-backend = "flit_core.buildapi"
```

Comment:
Can we inline these into the tests (by writing out the `pyproject.toml` to the temporary directory at the start of the test)? I would find it easier to write and maintain, and it matches the pattern we use elsewhere.

Reply:
I'll inline these last when everything else is done; inline test make debugging harder than it needs to be.

Reply:
That's fine, thanks.